### PR TITLE
Fix #65161 repeatlist unwind section by section UPDATED

### PR DESCRIPTION
**Pull request by ericfont**
This PR is able to unwind the repeat list for scores with multiple sections separated by SECTION_BREAKS.  This is updated version of PR #2165 and incorprates the latest fixes to repeat list which I needed to for this to pass. I created a massive test file repeat36.mscx from album of the entire set of tests repeat01 through repeat35, except for repeat13.mscx.

I did not include repeat13.mscx because it has DC al Coda without a corresponding To Coda and Coda defined.  My imlementation first searches for markers inside the current section, and then if not found will search rest of score as a fallback.  However, I do not believe jumping to markers in outside sections should be officially supported behavior, so I didn't include repeat13 in the joined album test.

### CODE REVIEW

**Comment by ericfont** [comment]
wait...travis didn't pass my repeat test, and it didn't squash...let me correct this...

**Comment by ericfont** [comment]
I'm actually confused why travis is failing...I'm unable to reproduce the fail when I playback.  I might have to try to actually run make  test & debug that.

**Comment by ericfont** [comment]
ok. the test passed.  My error a couple hours ago was that when reach a Fine, I was finalizing the repeat segment, TWICE.  So I had an extra set of the last repeat segment in the repeat list.  Anyway, since this updated pull request is working and has the latest fixes incorporated, I'm going to cancel my previous pull request.  So this should be ready to merge.

File: libmscore/repeatlist.cpp [hunk 1]
```diff
@@ -260,19 +284,49 @@ void RepeatList::unwind()
             return;
 
 // qDebug("unwind===================");
+
+      for (Measure* m = fm; m; m = m->nextMeasure())
+            m->setPlaybackCount(0);
+
+      // partition score by section breaks, and unwind individual sections seperately
```
Comment (ericfont) [inline libmscore/repeatlist.cpp]: Wait.  Regarding what @MarcSabatella said to me yesterday in the chat, a frame is not always a measure.  So I have a small bug when I partition score by section breaks here if a section break occures on a non-measure measure-base, for example the horizontal frame here:
section_break_on_non-measure_frame

All this bug does is ignore the section break entirely (not a crash or anything).  So let me fix it here.

I also notice that a section-break on a frame will not incur a pause during playback.  That is a seperate bug I will file in musescore issue tracker.

Comment (ericfont) [inline libmscore/repeatlist.cpp]: I also notice that section break on a frame will not reset the starting measure of next section to 1

**Comment by ericfont** [comment]
My latest push modifies unwind() to handles cases where section breaks occur on non-measure MeasureBase objects, like vertical/horizontal frames, and invokes unwindSection with actual measures only.

**Comment by lasconic** [comment]
Can you add some tests for these cases too? thanks!

**Comment by ericfont** [comment]
I do have test case repeat37 that I was working with, that I've ensured playbacks correctly, however, my new test repeat37 will not pass travis test because the numbering scheme is messed up if a section break occurs on a non-measure frame: see issue https://musescore.org/en/node/73801.  I asked if there were any takers on the chat, but no one did.  That would need to be fixed before travis passes my repeat37.

**Comment by lasconic** [comment]
OK let me take a look.

**Comment by ericfont** [comment]
I've self assigned that.  Here is my repeat test w/ expected behavior labeled (but I didn't include in git): https://musescore.org/sites/musescore.org/files/issues/repeat37.mscx

**Comment by ericfont** [comment]
(I'm making a note that because it is not possible to add segnos to non-measure frames, it is not necessary for Score::searchLabel() or Score::searchLabelWithinSectionFirst() to expand their search to include non-measure frames.)

**Pull request merged**